File: test_rpc.py
import os
from supabase import create_client, Client
from dotenv import load_dotenv

load_dotenv("backend/.env")

url = os.environ.get("SUPABASE_URL")
key = os.environ.get("SUPABASE_SERVICE_ROLE_KEY")

supabase = create_client(url, key)

try:
    res = supabase.rpc('exec_sql', {'sql': 'SELECT 1'}).execute()
    print("SUCCESS: Result:", res.data)
except Exception as e:
    print("FAILED:", e)

File: test_rpc.js
import { createClient } from '@supabase/supabase-js'
import dotenv from 'dotenv'
import path from 'path'

dotenv.config({ path: 'backend/.env' })

const supabaseUrl = process.env.SUPABASE_URL
const supabaseKey = process.env.SUPABASE_SERVICE_ROLE_KEY

const supabase = createClient(supabaseUrl, supabaseKey)

async function test() {
    try {
        const { data, error } = await supabase.rpc('exec_sql', { sql: 'SELECT 1' })
        if (error) {
            console.log("FAILED:", error)
        } else {
            console.log("SUCCESS: Result:", data)
        }
    } catch (e) {
        console.log("ERROR:", e)
    }
}

test()
